toolbox/library/src/YarpRpc.cpp

```diff
@@ -0,0 +1,257 @@
+    ok = ok && m_parameters.getParameter("PortName", pImpl->serverPortName);
+    ok = ok && m_parameters.getParameter("RpcCommand", command);
+
+    if (!ok) {
+        bfError << "Failed to read input parameters.";
+        return false;
+    }
+
+    // CLASS INITIALIZATION
+    // ====================
+
+    pImpl->network = std::make_unique<yarp::os::Network>();
+
+    if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        bfError << "YARP server wasn't found active.";
+        return false;
+    }
+
+    // Open a rpc client port
+    pImpl->clientPortName = "...";
+
+    if (!pImpl->rpcClientPort.open(pImpl->clientPortName)) {
+        bfError << "Error while opening yarp rpc server port.";
+        return false;
+    }
+
+    if (!yarp::os::Network::connect(pImpl->rpcClientPort.getName(), pImpl->serverPortName)) {
+        bfError << "Failed to connect " << pImpl->serverPortName << " to "
+                << pImpl->rpcClientPort.getName() << ".";
+        return false;
+    }
+
+    // Store rpc command in bottle
+    pImpl->rpcCommand.fromString(command);
+
+    return true;
+}
+
+bool YarpRpc::terminate(const BlockInformation* /*blockInfo*/)
+{
+    if(pImpl->rpcClientPort.isWriting()) {
+        bfWarning << "Interrupting rpc port";
+        pImpl->rpcClientPort.interrupt();
+    }
+
+    yarp::os::Network::disconnect(pImpl->rpcClientPort.getName(), pImpl->serverPortName);
+
+    // Close the port
+    pImpl->rpcClientPort.close();
+
+    return true;
+}
+
+bool YarpRpc::output(const BlockInformation* blockInfo)
+{
+    InputSignalPtr triggerSignal = blockInfo->getInputPortSignal(InputIndex::TriggerSignal);
+
+    if (!triggerSignal) {
+        bfError << "Input signal not valid.";
+        return false;
+    }
+
+    bool rpcStatus = true;
+
+    //TODO: Update to getting bool type after
+    //testing boolean support more thoroughly
+    double trigger = triggerSignal->get<double>(0);
+
+    if (pImpl->risingEdgeTrigger) {
+
+        if (trigger && !pImpl->previousTrigger) {
+
+            rpcStatus = sendRpcCommand();
+
+        }
+
+    }
+    else if (static_cast<bool>(trigger)) {
+
+        rpcStatus = sendRpcCommand();
+
+    }
+
+    //Update previous trigger
+    pImpl->previousTrigger = trigger;
+
+    //TODO: Check if it is useful to forward reponse to an output signal
+    return rpcStatus;
+}
+
+bool YarpRpc::sendRpcCommand()
+{
+    if (pImpl->rpcClientPort.isWriting()) {
+        bfError << "YarpRpc client port is blocked in writing";
+        return false;
+    }
+
+    yarp::os::Bottle response;
+
+    // Initialize the time counter for the timeout
+    const double t0 = yarp::os::SystemClock::nowSystem();
+
+    while(!pImpl->rpcClientPort.write(pImpl->rpcCommand, response)) {
+
+        yarp::os::Time::delay(0.0005);
+        const double now = yarp::os::SystemClock::nowSystem();
+
+        if((now - t0) > pImpl->timeout) {
```

Comment:
In the case of `YarpRead`, the timeout is a mask parameter. Not a blocking comment, but if this value is hardcoded as it is I would create a constant variable in this cpp file to store the timeout value.